### ISSUE TITLE
AncestorSampleMixin does not use common_variants=False optimisation

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#699]
When we request get_annotation_kwargs() up the parent chain to get a cohort genotype, we pass up  "common_variants=False" if a population filter removes common variants

But AncestorSampleMixin adds the filtered relation to its cohort genotype via get_annotation_kwargs (in case it wasn't present and we need it) but this doesn't include the "common_variants=False" so will always bring them down

Basically it doesn't take advantage of the optimisation

**Comment by davmlaw** [comment]
In case I don't take the rest of the merge_node_optimise branch, I did the work there under commit 0795c04f

```
# analysis.models.nodes.analysis_node.AnalysisNode.get_annotation_kwargs
# Just above line a_kwargs.update(self._get_annotation_kwargs_for_node(**kwargs))
kwargs["existing_annotation_kwargs"] = set(a_kwargs.keys())
```


```
# analysis.models.nodes.cohort_mixin.SampleMixin._get_annotation_kwargs_for_node
# add at top of method
kwargs["override"] = False
```


```
# snpdb.models.models_cohort.CohortGenotypeCollection.get_annotation_kwargs
    def get_annotation_kwargs(self, **kwargs) -> Dict:
        """ For Variant.objects.annotate """
        annotation_kwargs = {}
        already_set = self.cohortgenotype_alias in kwargs.get("existing_annotation_kwargs", set())
        if not (already_set and kwargs.get("override") is False):
            collections = [self.pk]
            if kwargs.get("common_variants", True):
                collections.append(self.common_collection_id)
            cgc_condition = Q(cohortgenotype__collection__in=collections)
            annotation_kwargs[self.cohortgenotype_alias] = FilteredRelation('cohortgenotype', condition=cgc_condition)
        return annotation_kwargs
```

**Comment by davmlaw** [comment]
Merged in with rest of merged node changes, deployed to VG.com
